cms/djangoapps/contentstore/views/course.py

```diff
@@ -255,13 +255,13 @@ def course_handler(request, course_key_string=None):
         response_format = request.GET.get('format') or request.POST.get('format') or 'html'
         if response_format == 'json' or 'application/json' in request.META.get('HTTP_ACCEPT', 'application/json'):
             if request.method == 'GET':
-                course_key = CourseKey.from_string(course_key_string)
```

Comment:
We should grab the `course_key` just once in this view at the top :). that way we can avoid extra database calls if we're just going to return a 404 anyway